File: Assignment 03 (Recursion, Heuristic Functions)/Prolog/Assignment3_1.pl
sum1(A,D,1,A):-
               !.

sum1(A,D,N,S):-
               N1 is N-1, A1 is A, D1 is D, sum1(A1,D1,N1,S1),
               S is S1+A1+(N-1)*D1.

File: Assignment 03 (Recursion, Heuristic Functions)/Python/Assignment 3_1.py
def sum1(N, I, F):
    if N == 0:
        return 0
    elif N >= 1:
        return sum1(N - 1, I, F) + F + (N - 1) * I


f = int(input("First Number: "))
d = int(input("Interval: "))
n = int(input("Total Terms: "))

print("Series Sum: ", sum1(n, d, f))
